### PR TITLE
fix(argo-workflows): Align version label

### PR DESCRIPTION
**Pull request by yu-croco**
resolves https://github.com/argoproj/argo-helm/issues/2335

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/templates/controller/workflow-controller-service.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: {{ .Release.Namespace | quote }}
   labels:
     {{- include "argo-workflows.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
-    app.kubernetes.io/version: {{ default (include "argo-workflows.defaultTag" .) .Values.controller.image.tag | trunc 63 | quote }}
+    app.kubernetes.io/version: {{ include "argo-workflows.controller_chart_version_label" . }}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/controller/workflow-controller-service.yaml:9]: 📝  `app.kubernetes.io/version` showed correct value as expected with the following values.
```yaml
controller:
  serviceLabels:
    testing: "1234"
  image:
    registry: 1234.dkr.ecr.us-west-2.amazonaws.com
    repository: third-party/argoproj/workflow-controller
    tag: v3.5.0@sha256:dda36f7a872f463a68df04d72755c086eb193fa1645e1f099f1670dd5f99a277
  metricsConfig:
    enabled: true
server:
  image:
    registry: 1234.dkr.ecr.us-west-2.amazonaws.com
    repository: third-party/argoproj/argocli
    tag: v3.5.0@sha256:608b6d532b6a1ad4ae82b1e05ee844eb2f4fffe90a8524421f95e9355a7e4be1
  secure: true
  authMode: sso
```
image

**Review by jmeridth — APPROVED** [review]

File: charts/argo-workflows/Chart.yaml [hunk 2]
```diff
@@ -16,5 +16,5 @@ annotations:
     fingerprint: 2B8F22F57260EFA67BE1C5824B11F800CD9D2252
     url: https://argoproj.github.io/argo-helm/pgp_keys.asc
   artifacthub.io/changes: |
-    - kind: added
-      description: Add parameters for tuning revisionHistoryLimit and emptyDir volumes.
+    - kind: fixed
+      description: Aligne version label
```
Comment (tico24) [inline charts/argo-workflows/Chart.yaml:20]: nit: it's "align"

Comment (yu-croco) [reply charts/argo-workflows/Chart.yaml:20]: let me resign an engineer. 👴

Comment (tico24) [reply charts/argo-workflows/Chart.yaml]: You can resign after there are no more PRs needed.

Comment (yu-croco) [reply charts/argo-workflows/Chart.yaml]: I fixed in https://github.com/argoproj/argo-helm/pull/2342/commits/d84f766889a39c9b08c82031271fdf4cf39e4205 🙋

**Review by tico24 — APPROVED** [review]

File: charts/argo-workflows/Chart.yaml (hunk 2 above)
Comment (tico24) [reply charts/argo-workflows/Chart.yaml]: lgtm!

**Pull request merged**